package/MDAnalysis/core/selection.py

```diff
@@ -838,26 +858,48 @@ def apply(self, group):
         return group[mask].unique
 
 
-class ResnumSelection(RangeSelection):
-    token = 'resnum'
-    field = 'resnums'
+class FloatRangeSelection(RangeSelection):
+    """Range selection for float values"""
 
+    pattern = f"({FLOAT_PATTERN}){RANGE_PATTERN}({FLOAT_PATTERN})"
+    dtype = float
 
-class ByNumSelection(RangeSelection):
-    token = 'bynum'
-    field = 'indices'
-    value_offset = 1  # queries are in 1 based indices
+    def apply(self, group):
+        mask = np.zeros(len(group), dtype=bool)
+        vals = getattr(group, self.field) + self.value_offset
 
+        for upper, lower in zip(self.uppers, self.lowers):
+            if upper is not None:
+                thismask = vals > lower
+                thismask &= vals < upper
+                # thismask |= np.isclose(vals, lower, atol=self.atol,
+                #                        rtol=self.rtol)
+                # thismask |= np.isclose(vals, upper, atol=self.atol,
+                #                        rtol=self.rtol)
```

Comment:
Thoughts on this? The integer range selection includes the ends; should float range selections?

Reply:
Yes, then let's be consistent. (The clearer version for floats would be strict inequalities, i.e., an open interval, but my feeling is that users rather go for "this value or more" or "this value or less" and don't want to include their own epsilon... "this value minus epsilon or more".)

Reply:
I am going back and forth on using `isclose()` for the endpoints. 

But I am now tending to just use `<=` and `>=` and then it behaves exactly as normal comparison operators in Python. I think that's less confusing than the special case where we would make the endpoints fuzzy.

Other opinions welcome! – @jbarnoud @richardjgowers @zemanj ?

Reply:
I like that solution, but I've kept it in a separate commit (https://github.com/MDAnalysis/mdanalysis/pull/2927/commits/fc68f03c26aeb55c10f1f6797e67319f4765337e) in case people disagree :-)